pkg/cloudprovider/types.go

```diff
@@ -83,6 +88,20 @@ type InstanceType struct {
 
 type InstanceTypes []*InstanceType
 
+func (its InstanceTypes) FilterByMinValues(reqs scheduling.Requirements) (InstanceTypes, error) {
+	if !reqs.HasMinValues() {
+		return its, nil
+	}
+
+	// We would have already filtered the invalid NodeClaim not meeting the minimum requirements in simulated scheduling results.
+	// Here the instanceTypeOptions changed again based on the price and requires re-validation.
+	if _, err := its.SatisfiesMinValues(reqs); err != nil {
```

Comment:
This function is so short, do we event need the helper? Also, it's just validating that minValues are satisfied if they exist based on the SatisfiesMinValues function. there's no actual filtering that's occurring here from what I can tell